qir/qat/Passes/TargetQisMappingPass/TransformationRulesPassConfiguration.cpp

```diff
@@ -0,0 +1,267 @@
+
+void TargetQisMappingPassConfiguration::setup(ConfigurationManager& config)
+{
+    config.setSectionName("Target QIS mapping", "Modification of the IR to adapt a specific QIS.");
+
+    // Experimental settings
+    config.addExperimentalParameter(delete_dead_code_, true, false, "delete-dead-code", "Deleted dead code.");
+    config.addExperimentalParameter(
+        clone_functions_, true, false, "clone-functions", "Clone functions to ensure correct qubit allocation.");
+    config.addExperimentalParameter(
+        transform_execution_path_only_, true, false, "transform-execution-path-only",
+        "Transform execution paths only.");
+
+    config.addExperimentalParameter(
+        max_recursion_, max_recursion_, uint64_t(1), "max-recursion",
+        "Defines the maximum recursion when unrolling the execution path");
+
+    config.addExperimentalParameter(
+        assume_no_exceptions_, false, false, "assume-no-except",
+        "Assumes that no exception will occur during runtime.");
+
+    config.addParameter(reuse_qubits_, false, "reuse-qubits", "Use to define whether or not to reuse qubits.");
+    config.addParameter(reuse_results_, false, "reuse-results", "Use to define whether or not to reuse results.");
+
+    // Rules
+    config.addParameter(
+        optimize_result_one_, true, "optimize-result-one",
+        "Maps branching based on quantum measurements compared to one to base adaptor "
+        "type measurement.");
+    config.addParameter(
+        optimize_result_zero_, true, "optimize-result-zero",
+        "Maps branching based on quantum measurements compared to zero to base adaptor "
+        "type measurement.");
+    config.addParameter(
+        optimize_result_comparison_, true, "optimize-result-comparison",
+        "Maps branching based on quantum measurements compared to each other to base adaptor "
+        "type measurement.");
+    config.addParameter(
+        use_static_qubit_array_allocation_, "use-static-qubit-array-allocation",
+        "Maps allocation of qubit arrays to static array allocation.");
+    config.addParameter(
+        use_static_qubit_allocation_, "use-static-qubit-allocation", "Maps qubit allocation to static allocation.");
+    config.addParameter(
+        use_static_result_allocation_, "use-static-result-allocation", "Maps result allocation to static allocation.");
+
+    config.addParameter(
+        disable_reference_counting_, true, "disable-reference-counting",
+        "Disables reference counting by instruction removal.");
+    config.addParameter(
+        disable_alias_counting_, true, "disable-alias-counting", "Disables alias counting by instruction removal.");
+    config.addParameter(
+        disable_string_support_, true, "disable-string-support", "Disables string support by instruction removal.");
+    config.addParameter(
+        disable_record_output_support_, true, "disable-record-output-support",
+        "Disables record output support by instruction removal.");
```

Comment:
Is it simply removing these calls? How would a program produce any kind of output then if this is set to true? And how come this is set to true by default?

Reply:
This was a request from Steffan (if I recall correctly): I believe that one or more of the providers just do a single measurement at the end and that all record functions are disregarded.